test/mdc2test.c

```diff
@@ -41,59 +35,54 @@ static unsigned char pad2[16] = {
     0x35, 0xD8, 0x7A, 0xFE, 0xAB, 0x33, 0xBE, 0xE2
 };
 
-int main(int argc, char *argv[])
+static int test_mdc2(void)
 {
-    int ret = 1;
+    int testresult = 0;
     unsigned char md[MDC2_DIGEST_LENGTH];
-    int i;
     EVP_MD_CTX *c;
     static char text[] = "Now is the time for all ";
+    size_t tlen = strlen(text);
 
 # ifdef CHARSET_EBCDIC
-    ebcdic2ascii(text, text, strlen(text));
+    ebcdic2ascii(text, text, tlen);
 # endif
 
     c = EVP_MD_CTX_new();
-    if (c == NULL
-        || !EVP_DigestInit_ex(c, EVP_mdc2(), NULL)
-        || !EVP_DigestUpdate(c, (unsigned char *)text, strlen(text))
-        || !EVP_DigestFinal_ex(c, &(md[0]), NULL))
-        goto err;
-
-    if (memcmp(md, pad1, MDC2_DIGEST_LENGTH) != 0) {
-        for (i = 0; i < MDC2_DIGEST_LENGTH; i++)
-            printf("%02X", md[i]);
-        printf(" <- generated\n");
-        for (i = 0; i < MDC2_DIGEST_LENGTH; i++)
-            printf("%02X", pad1[i]);
-        printf(" <- correct\n");
-        goto err;
-    } else {
-        printf("pad1 - ok\n");
-    }
-
-    if (!EVP_DigestInit_ex(c, EVP_mdc2(), NULL))
-        goto err;
+    if (!TEST_ptr(c))
+        goto end;
+    if (!TEST_true(EVP_DigestInit_ex(c, EVP_mdc2(), NULL)))
+        goto end;
+    if (!TEST_true(EVP_DigestUpdate(c, (unsigned char *)text, tlen)))
+        goto end;
+    if (!TEST_true(EVP_DigestFinal_ex(c, &(md[0]), NULL)))
```

Comment:
It would be more consistent with our style elsewhere to put these all into a single if with one "goto end" (similar to the original version).